probe-rs/src/probe.rs

```diff
@@ -953,7 +953,30 @@ pub trait JTAGAccess: DebugProbe {
 
     /// Executes register writes scheduled by `schedule_write_register`
     /// If the probe doesn't support batching this will be emulated.
-    fn execute(&mut self) -> Result<Box<dyn CommandResults>, DebugProbeError>;
+    fn execute(&mut self) -> Result<Box<dyn CommandResults>, BatchExecutionError>;
+}
+
+#[derive(thiserror::Error, Debug)]
+pub struct BatchExecutionError {
+    pub error: DebugProbeError,
```

Comment:
For the best compatibility, this should either be renamed to `source`, or the `#[source]` attribute should be added.
```suggestion
    #[source]
    pub error: DebugProbeError,
```
With this, `thiserror` will provide an implementation of [`Error::source`].

[`Error::source`]: https://doc.rust-lang.org/std/error/trait.Error.html#method.source